### PR TITLE
Docs for connecting traces to profiles

### PR DESCRIPTION
**Pull request by kayayarai**
### What does this PR do?

Adds docs for Connecting Traces to Profiles

### Motivation
DOCS-1477

### Preview

https://docs-staging.datadoghq.com/kari/code-hotspots/tracing/profiler/connect_traces_and_profiles/
https://docs-staging.datadoghq.com/kari/code-hotspots/tracing/profiler/

---

### Reviewer checklist
- [ ] Review the changed files.
- [ ] Review the URLs listed in the [Preview](#preview) section.
- [ ] Review any mentions of "Contact Datadog support" for internal support documentation.

### CODE REVIEW

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+If your application is showing performance problems in production, it's helpful to connect the distributed trace information from APM with the full profile of the code stack. Application processes that have both APM distributed tracing and continuous profiler enabled are automatically linked, so you can move directly from span information to profiling data on the Code Hotspots tab, and find specific lines of code related to performance issues.
+
+{{< img src="tracing/profiling/code_hotspots_tab.gif" alt="Code Hotspots tab shows profiling information for a APM trace span">}}
+
+## Prerequisites
+
+{{< programming-lang-wrapper langs="java,python" >}}
+{{< programming-lang lang="java" >}}
+For manually instrumented code, Continuous Profiler requires Scope activation of spans:
```
[the remark this answers is not in the document]
Comment (GaurabAryal) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:24]: Also the python one says the library requirement version. I think we want to say 0.65.0+ here for version (although likely supported from before, I cannot find it in release logs)

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+The breakdown view on the left side is a list of types of time spent executing that span. Depending on the runtime and language, this list of types varies:
+
+- **Method durations** shows the overall time taken by each method from your code.
+- **CPU** shows the time taken executing CPU tasks.
+- **Synchronization** shows the time taken waiting for a lock of a synchronized object.
+- **Garbage collection** shows the time taken waiting for the garbage collector to execute.
+- **VM operations** (Java only) shows the time taken waiting for VM operations that are not related to garbage collection (for example, heap dumps).
+- **File I/O** shows the time taken waiting for a disk read/write operation to execute.
+- **Socket I/O** shows the time taken waiting for a network read/write operation to execute.
+- **Object wait** shows the time ...
```
Comment (GaurabAryal) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:56]: @hdafgard what should we put here?

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+Click on one of these types to see a corresponding list, ordered by time, of the methods that are taking time. Clicking on the plus `+` will expand the stack trace to that method **in reverse order**.
```
Comment (GaurabAryal) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:58]: I think we want to put this _after_ the `other` section

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+In the breakdown view, **Other** shows the time taken to execute the span that cannot be explained by profiling data. It is not uncommon to have a small amount of unexplained time (less than 10%). Potential reasons for Other time include:
+
+  - The span you selected isn't directly mapped to any execution. Profiling data is associated uniquely to spans when they are executing on a specific thread. For example, some spans are created/used uniquely as virtual containers of a series of related processing steps and never actually directly associated with any thread execution.
+  - Your application process cannot access CPU resources to execute and is paused. There is no way for profiler to know about competing resources from other processes or containers.
+  - The application is locked in synchronization or in I/O events that are individually lower than 10ms: Java profiler gets data for paused thread events (locks, I/O, parks) that are larger than 10ms. If you want to reduce that threshold, see [link][].
```
Comment (GaurabAryal) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:64]: I'll add the link here shortly when the troubleshooting section gets merged!

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  - The span you selected is short. Profiling is a sampling mechanism that regularly looks at how your code behaves. There might not be enough representative data for spans shorter than 50ms
+  - Missing instrumentation: Profiling breakdown requires that spans are associated with executing threads by activating these spans in the ScopeManager. Some custom instrumentations don't activate these spans properly and so we can't map them to executing threads. If this span comes from a custom integration you can check [link][] for information on how to improve this.
```
Comment (GaurabAryal) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:66]: @hugokash what's the link here?

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+
+## View profile from trace
```
Comment (GaurabAryal) [inline content/en/tracing/profiler/connect_traces_and_profiles.md:68]: @kayayarai this is missing the second gif that I sent you where we click the 'View Profile' link and it goes to the flame graph view.

**Review by GaurabAryal — CHANGES_REQUESTED** [review]
Few comments left for Hugo/Kari/Henrik

File: content/en/tracing/profiler/connect_traces_and_profiles.md (hunk 3 above)
Comment (kayayarai) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:59]: I kinda think it's nice having this close to the list of types, so instead, I added "Other" to the list, left this here, and then continued with the extended description of what Other might mean.

File: content/en/tracing/profiler/connect_traces_and_profiles.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
+---
+title: Connect Traces and Profiles
```
[the remark this answers is not in the document]
Comment (hugokash) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:2]: agreed

File: content/en/tracing/profiler/connect_traces_and_profiles.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (hugokash) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:24]: yes, agreed

File: content/en/tracing/profiler/connect_traces_and_profiles.md (hunk 5 above)
Comment (hugokash) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:67]: https://docs.datadoghq.com/tracing/setup_overview/custom_instrumentation/java#manually-creating-a-new-span

File: content/en/tracing/profiler/connect_traces_and_profiles.md (hunk 2 above)
Comment (hdafgard) [reply content/en/tracing/profiler/connect_traces_and_profiles.md:56]: It's related to the Synchronization data, so something like 
`**Object wait** shows the time taken waiting for a notify call on an object.` or something along those lines.

https://docs.oracle.com/en/java/javase/15/docs/api/java.base/java/lang/Object.html#wait()

**Pull request closed without merging**